sdcm/utils/thread.py

```diff
@@ -0,0 +1,34 @@
+import sys
+from functools import wraps
+import os
+import signal
+
+
+def background_failure_handler(signum, frame):  # pylint: disable=unused-argument
+    raise Exception("Failure/Error in background threads")
```

Comment:
If the thread is executed by a function which has retrying modifier, the Exception will be suppressed and continue retrying?

Reply:
yes, the exceptions handling is done at the level of thread creation, hence anything caught inside the code won't be caught